### PR TITLE
Basic EMR client for Spark Jobs

### PR DESCRIPTION
**Pull request by huitseeker**
## What changes were proposed in this pull request?
Adds a basic EMR client 

## How was this patch tested?
Manually.

### CODE REVIEW

**Comment by saudet** [comment]
With samediff we do now...

**Comment by wmeddie** [comment]
Ah ok.

**Comment by huitseeker** [comment]
This PR does not introduce a Java8 dependency. The `Optional` used in this PR is not the Java8 `Optional`.

File: deeplearning4j-scaleout/deeplearning4j-aws/pom.xml [hunk 1]
```diff
@@ -46,6 +46,11 @@
                         <artifactId>jsch</artifactId>
                         <version>0.1.51</version>
                 </dependency>
+            <dependency>
+                <groupId>org.apache.spark</groupId>
+                <artifactId>spark-core_2.11</artifactId>
+                <version>2.1.0</version>
```
Comment (agibsonccc) [inline deeplearning4j-scaleout/deeplearning4j-aws/pom.xml]: Don't we want a ${spark.version} here? That or at least use an inherited version somewhere? Would rather not have multiple spark version declarations.

File: deeplearning4j-scaleout/deeplearning4j-aws/src/main/java/org/deeplearning4j/aws/emr/EmrConfig.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    Configuration toAwsConfig() {
+        Configuration config = new Configuration().withClassification(classification).withProperties(properties);
+        List<Configuration> subConfigs = new ArrayList<>();
+        for (EmrConfig conf : configs){
+            subConfigs.add(conf.toAwsConfig());
+        }
+        return config.withConfigurations(subConfigs);
+    }
+
+    protected String classification;
```
Comment (agibsonccc) [inline deeplearning4j-scaleout/deeplearning4j-aws/src/main/java/org/deeplearning4j/aws/emr/EmrConfig.java]: Fields at top is normal convention?

File: deeplearning4j-scaleout/deeplearning4j-aws/src/main/java/org/deeplearning4j/aws/emr/S3Url.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Getter
+    private String bucket;
+    @Getter
+    private String key;
+
+    public S3Url(String url){
+        if (!url.startsWith("s3://") || url.length() < 5){
+            throw new IllegalArgumentException("S3 URL should start with s3://");
+        }
```
Comment (agibsonccc) [inline deeplearning4j-scaleout/deeplearning4j-aws/src/main/java/org/deeplearning4j/aws/emr/S3Url.java]: Could we use URI.create/parse here instead?

**Pull request merged**